memory/table.go

```diff
@@ -458,13 +484,35 @@ func (i *tableIter) Next(ctx *sql.Context) (sql.Row, error) {
 	return row, nil
 }
 
-func (i *tableIter) colIsProjected(idx int) bool {
-	for _, colIdx := range i.columns {
-		if idx == colIdx {
-			return true
+// normalizeRowForRead returns a copy of the row with nil values inserted for any virtual columns
+func normalizeRowForRead(row sql.Row, numColumns int, columns []int, virtualCols []int) sql.Row {
+	if len(virtualCols) == 0 {
+		return row
+	}
+
+	virtualRow := make(sql.Row, numColumns)
+
+	var j int
+	if len(columns) != 0 {
+		virtualRow = make(sql.Row, len(columns))
```

Comment:
bit confused here. Why do we alloc virtualRow twice in the general case? And what does `columns[i] == virtualCols[j]` mean? I thought these would have been non-overlapping positional ordinals of regular vs virtual columns

Reply:
Might be some cleanup there from an earlier attempt, i'll take a pass